### PR TITLE
WiP json: templatize json streamer

### PR DESCRIPTION
**Pull request by jmarantz**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by jmarantz** [comment]
@wbpcode FYI -- we might not use this.

**Comment by jmarantz** [comment]
@wbpcode PTAL for a mechanism to share json serialization logic between contexts where we want to build a buffer vs output a small string.

**Comment by wbpcode** [comment]
I will prefer to split the basic serialization method out from the Streamer to make the code more clear. 🤔

File: source/common/json/json_streamer.h [hunk 1]
```diff
@@ -6,20 +6,47 @@
...
+  StringTemplate(std::string& buffer) : string_(buffer) {}
+
+  void add(absl::string_view a) { absl::StrAppend(&string_, a); }
+  void add(absl::string_view a, absl::string_view b, absl::string_view c) {
+    absl::StrAppend(&string_, a, b, c);
+  }
+
+private:
+  std::string& string_;
+};
```
Comment (wbpcode) [inline source/common/json/json_streamer.h:40]: I will think single `addFragments` is good enough for this scenario. By this way, although we cannot take the advantage of the `absl::StrAppend`. But it is fine from my previous benchmark results.
And note, if string is used, I assume it's used for short JSON serialization and we will pre-allocated enough buffer for the string, `absl::StrAppend` make no sense in this case.


```

using StringTemplate = Envoy::Buffer::Instance;

class StringTemplate {
public:
  StringWriter(size_t initial_buffer_size = 2048) { buffer_.reserve(initial_buffer_size); }

  void addFragments(absl::Span<const absl::string_view> fragments) {
    for (absl::string_view fragment : fragments) {
      buffer_.append(fragment.data(), fragment.size());
    }
  }
  std::string buffer_;
};

```

**Pull request closed without merging**